src/installer/corehost/cli/apphost/bundle/header.h

```diff
@@ -27,7 +32,31 @@ namespace bundle
         uint32_t minor_version;
         int32_t num_embedded_files;
 
-        bool is_valid() const;
+        bool is_valid(bool exact_match = false) const;
+    };
+#pragma pack(pop)
+
+    enum header_flags_t : uint64_t
+    {
+        none = 0,
+        needs_extraction = 1,
```

Comment:
This is effectively redundant information, right? We could compute this from the file types - just like the bundler does it.
What will be the use of this? Do we need this to help with simplifying `apphost` or `hostfxr`?

Reply:
Yes, it is redundant information. In fact, everything in V2-header is redundant. 
Caching the `deps.json` and `runtimeconfig.json` information is necessary to simplify calling the `hostfxr`. 

I added `NeedsExtraction` with the same idea, but it may not be necessary, because `hostpolicy`, which perform the extraction, will need to iterate over all `FileEntry`s anyway.

Reply:
Good point about the other fields. Worth a comment in the code - especially the "why".